README.md

```diff
@@ -1 +1,218 @@
+// exposes should be idempotent.
+service Controller {
+  // EnableVolumeReplication RPC call to enable the volume replication.
+  rpc EnableVolumeReplication (EnableVolumeReplicationRequest)
+  returns (EnableVolumeReplicationResponse) {}
+  // DisableVolumeReplication RPC call to disable the volume replication.
+  rpc DisableVolumeReplication (DisableVolumeReplicationRequest)
+  returns (DisableVolumeReplicationResponse) {}
+  // PromoteVolume RPC call to promote the volume.
+  rpc PromoteVolume (PromoteVolumeRequest)
+  returns (PromoteVolumeResponse) {}
+  // DemoteVolume RPC call to demote the volume.
+  rpc DemoteVolume (DemoteVolumeRequest)
+  returns (DemoteVolumeResponse) {}
+  // ResyncVolume RPC call to resync the volume.
+  rpc ResyncVolume (ResyncVolumeRequest)
+  returns (ResyncVolumeResponse) {}
+}
+```
+
+
+#### EnableVolumeReplication
+
+```protobuf
+// EnableVolumeReplicationRequest holds the required information to enable
+// replication on a volume.
+message EnableVolumeReplicationRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 3 [(replication_secret) = true];
+}
+
+// EnableVolumeReplicationResponse holds the information to send when
+// replication is successfully enabled on a volume.
+message EnableVolumeReplicationResponse {
+}
+```
+
+
+#### DisableVolumeReplication
+
+```protobuf
+// DisableVolumeReplicationRequest holds the required information to disable
+// replication on a volume.
+message DisableVolumeReplicationRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 3 [(replication_secret) = true];
+}
+
+// DisableVolumeReplicationResponse holds the information to send when
+// replication is successfully disabled on a volume.
+message DisableVolumeReplicationResponse {
+}
+```
+
+
+#### PromoteVolume
+
+```protobuf
+// PromoteVolumeRequest holds the required information to promote volume as a
+// primary on local cluster.
```

Comment:
local cluster ?

Reply:
local cluster means the second cluster where it is not primary and should be run from that itself locally.

Reply:
got it.